[ONP-949] - Add documentation for overriding the default entrypoint (#10482)
circleci/circleci-docs: docs/guides/modules/execution-runner/pages/container-runner.adoc

    command: ["--host=unix:///var/run/docker.sock"]
```

> [#custom-entrypoint]

Reviewer: ```suggestion
```

Author: Any reason to remove this? I thought this was so you could hook it in a link.

Reviewer: You can but it will just create an ID with the full section name  (#override-the-default-container-entrypoint) if you don't configure one. If there's a reason to give the section an ID that's different from the actual section name you can keep it!

> == Overriding the default container entrypoint

Reviewer: ```suggestion
== Override the default container entrypoint
```

NOTE: Overriding the default container entrypoint requires container runner version 3.1.10 or later.

WARNING: Overriding the default container entrypoint is an advanced operation. An incorrectly configured entrypoint can result in zombie processes, failed signal propagation, and unclean container shutdown.

By default, the task agent (GOAT) runs as PID 1 in the primary task container. You can override this by specifying a custom entrypoint in either the CircleCI job configuration or the resource class pod template spec.

Any custom entrypoint must fulfill the following requirements:

* Reap zombie processes and clean up child processes.
* Propagate signals to child processes, including GOAT.
* Manage the container lifecycle to ensure a clean exit.
* Spawn GOAT as a child process using the automatically provided command and arguments.

link:https://github.com/krallin/tini[Tini] is a purpose-built container init system that satisfies these requirements with minimal configuration.

=== CircleCI job configuration

> Set a custom entrypoint using the `entrypoint` field on the Docker image in your CircleCI `config.yml`. Container runner automatically passes the GOAT command and arguments to your entrypoint:

Reviewer: ```suggestion
Set a custom entrypoint using the `entrypoint` field on the Docker image in your CircleCI configuration file. Container runner automatically passes the GOAT command and arguments to your entrypoint:
```

[source,yaml]
----
jobs:
  test:
    docker:
      - image: krallin/ubuntu-tini
        entrypoint: ["/usr/bin/tini", "--"] <1>
    resource_class: <your-resource-class-here>
    steps:
      - run:
          name: Who is PID 1?
          command: ps -p 1 -o comm=
----

> <1> The GOAT command and arguments are automatically appended and passed to `tini`.

Reviewer: ```suggestion
        entrypoint: ["/usr/bin/tini", "--"] # GOAT command and arguments automatically appended and passed to `tini`
    resource_class: <your-resource-class-here>
    steps:
      - run:
          name: Who is PID 1?
          command: ps -p 1 -o comm=
----

```

=== Resource class pod template spec

> To apply a custom entrypoint to all jobs using a resource class, configure it in the pod template spec within your Helm `values.yaml`. See <<resource-class-configuration-custom-pod,Resource class configuration and custom task pod configuration>> for details on pod template spec configuration. The same entrypoint requirements apply.

Reviewer: ```suggestion
To apply a custom entrypoint to all jobs using a resource class, configure it in the pod template spec within your Helm `values.yaml`. See <<resource-class-configuration-custom-pod>> for details on pod template spec configuration. The same entrypoint requirements apply.
```
